pytransform3d/batch_rotations.py

```diff
@@ -38,6 +38,52 @@ def norm_vectors(V, out=None):
     return out
 
 
+def norm_axis_angles(a):
+    """Normalize axis-angle representation.
+
+    Parameters
+    ----------
+    a : array-like, shape (..., 4)
+        Axis of rotation and rotation angle: (x, y, z, angle)
+
+    Returns
+    -------
+    a : array, shape (..., 4)
+        Axis of rotation and rotation angle: (x, y, z, angle). The length
+        of the axis vector is 1 and the angle is in [0, pi). No rotation
+        is represented by [1, 0, 0, 0].
+    """
+    a = np.asarray(a)
+
+    # Handle the case of only one axis-angle instance
+    only_one = a.ndim == 1
+    a = np.atleast_2d(a)
+
+    angles = a[..., 3]
+    norm = np.linalg.norm(a[..., :3], axis=-1)
+
+    zero_mask = (angles == 0.0) | (norm == 0.0)
+    non_zero_mask = ~zero_mask
+
+    res = np.empty_like(a)
```

Comment:
About the usage of `empty_like` which is callen on the input argument - are we sure that we want to use it without prior casting to float?

e.g. when `a` is an integer array, assignments `res[non_zero_mask, :3] = ...` are casted to int and may get output wrong results ...

However, I just tested it, and we luckily do have have an unintended "safeguard":

```python
a = np.array([[1, 2, 3, 2], [4, 5, 6, 3]])
# a = a.astype(float)  # works if uncommented

res = norm_axis_angles(a)

# crashes with:
#     res[negative_angle_mask, :3] *= -1.0
# numpy._core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'multiply' output from dtype('float64') to dtype('int64') with casting rule 'same_kind'
```

Maybe the type checking is something for the next PR? Probably this is an edge case ...

Reply:
I would avoid to make the decision in this pull request, since this should be handled consistently in the whole library.